shell/models/management.cattle.io.cluster.js

```diff
@@ -440,9 +447,12 @@ export default class MgmtCluster extends HybridModel {
     // cluster has the less human readable management cluster ID in it: fleet-default/c-khk48
 
     const verb = this.isLocal || isRKE1 || this.isHostedKubernetesProvider ? 'to' : 'from';
-    const from = `${ verb }Type`;
-    const id = `${ verb }Id`;
+    const res = findRelationship(verb, CAPI.RANCHER_CLUSTER, this.metadata?.relationships);
+
+    if (res) {
+      return res;
+    }
 
-    return this.metadata.relationships.find((r) => r[from] === CAPI.RANCHER_CLUSTER)?.[id];
+    return findRelationship(verb === 'to' ? 'from' : 'to', CAPI.RANCHER_CLUSTER, this.metadata?.relationships);
```

Comment:
not really a blocker but this line and 450 seem awfully similar to eachother especially since verb (which is the only value with a difference between the two of them) is set before either of them run. There might be a cleaner way to combine these logically.

Reply:
This one was tricky. There could be an ordered array that loops through checking each verb, exiting early if the first succeeds, but that would be more lines of code.